gcloud-java-dns/src/test/java/com/google/gcloud/dns/ITDnsTest.java

```diff
@@ -0,0 +1,1268 @@
+    assertEquals("1", created.id());
+    assertTrue(ImmutableList.of(ChangeRequest.Status.PENDING, ChangeRequest.Status.DONE)
+        .contains(created.status()));
+    assertEqChangesIgnoreStatus(created, DNS.getChangeRequest(ZONE1.name(), "1"));
+    checkChangeComplete(ZONE1.name(), "1");
+    DNS.applyChangeRequest(ZONE1.name(), CHANGE_DELETE_ZONE1);
+    checkChangeComplete(ZONE1.name(), "2");
+    // with options
+    created = DNS.applyChangeRequest(ZONE1.name(), CHANGE_ADD_ZONE1,
+        Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.ID));
+    assertTrue(created.additions().isEmpty());
+    assertNull(created.startTimeMillis());
+    assertTrue(created.deletions().isEmpty());
+    assertEquals("3", created.id());
+    assertNull(created.status());
+    checkChangeComplete(ZONE1.name(), "3");
+    DNS.applyChangeRequest(ZONE1.name(), CHANGE_DELETE_ZONE1);
+    checkChangeComplete(ZONE1.name(), "4");
+    created = DNS.applyChangeRequest(ZONE1.name(), CHANGE_ADD_ZONE1,
+        Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.STATUS));
+    assertTrue(created.additions().isEmpty());
+    assertNull(created.startTimeMillis());
+    assertTrue(created.deletions().isEmpty());
+    assertEquals("5", created.id());
+    assertNotNull(created.status());
+    checkChangeComplete(ZONE1.name(), "5");
+    DNS.applyChangeRequest(ZONE1.name(), CHANGE_DELETE_ZONE1);
+    checkChangeComplete(ZONE1.name(), "6");
+    created = DNS.applyChangeRequest(ZONE1.name(), CHANGE_ADD_ZONE1,
+        Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.START_TIME));
+    assertTrue(created.additions().isEmpty());
+    assertNotNull(created.startTimeMillis());
+    assertTrue(created.deletions().isEmpty());
+    assertEquals("7", created.id());
+    assertNull(created.status());
+    checkChangeComplete(ZONE1.name(), "7");
+    DNS.applyChangeRequest(ZONE1.name(), CHANGE_DELETE_ZONE1);
+    checkChangeComplete(ZONE1.name(), "8");
+    created = DNS.applyChangeRequest(ZONE1.name(), CHANGE_ADD_ZONE1,
+        Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.ADDITIONS));
+    assertEquals(CHANGE_ADD_ZONE1.additions(), created.additions());
+    assertNull(created.startTimeMillis());
+    assertTrue(created.deletions().isEmpty());
+    assertEquals("9", created.id());
+    assertNull(created.status());
+    // finishes with delete otherwise we cannot delete the zone
+    checkChangeComplete(ZONE1.name(), "9");
+    created = DNS.applyChangeRequest(ZONE1.name(), CHANGE_DELETE_ZONE1,
+        Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.DELETIONS));
+    checkChangeComplete(ZONE1.name(), "10");
+    assertEquals(CHANGE_DELETE_ZONE1.deletions(), created.deletions());
+    assertNull(created.startTimeMillis());
+    assertTrue(created.additions().isEmpty());
+    assertEquals("10", created.id());
+    assertNull(created.status());
+    checkChangeComplete(ZONE1.name(), "10");
+  }
+
+  @Test
+  public void testCreateChangeUsingZoneObject() {
```

Comment:
Nit: shouldn't be needed as we unit test Zone and have integration test for the service